detectron2/data/detection_utils.py

```diff
@@ -325,22 +325,23 @@ def annotations_to_instances_rotated(annos, image_size):
     return target
 
 
-def filter_empty_instances(instances, by_box=True, by_mask=True):
+def filter_empty_instances(instances, by_box=True, by_mask=True, box_threshold=0):
```

Comment:
Maybe it makes sense to use a slightly larger default like `1e-5` to avoid the issue you met.

Reply:
I agree. Would you want that default in Boxes.nonempty and RotatedBoxes.nonempty as well?

Reply:
The results of `Boxes.nonempty` may be used for many purposes, so it might be more natural to use 0 as default there.

However here the purpose is specifically about filtering bad boxes - and it's safe to say that boxes<1e-5 are by default "bad"

Reply:
Done